dask_sql/physical/rex/core/call.py

```diff
@@ -115,6 +115,25 @@ def __init__(self, operation: Callable, unary_operation: Callable = None):
 
     def reduce(self, *operands, **kwargs):
         if len(operands) > 1:
+            # Doing math against dates requires a Timedelta
+            # Find all the operands that are of a datetime64 type
+            date_operands = [
+                idx for idx in {0, 1} if pd.api.types.is_datetime64_dtype(operands[idx])
+            ]
+
+            # If there are datetime64 operands we need to make sure that the other operands
+            # in the list are Timedelta for the operation to work.
+            if date_operands:
+                # Operands is a Set, since we are altering it must convert to a List
+                operands = list(operands)
+
+                # Knowing there are datetime types in the operands check for incompatable other types
+                # If found, convert them to Timedelta
+                for idx, operand in enumerate(operands):
+                    # Default to `Day`/`D` since that is what PostgreSQL does
```

Comment:
Would probably add a `TODO` here bringing up the fact that it would be nice to push this scalar conversion up further so that we can assess if we are dealing with a `DATE` or `TIMESTAMP`.